### PR TITLE
Track logins and login attempts in the DB

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: app/models/login_activity.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class LoginActivity < ActiveRecord::Base
+  belongs_to :user, polymorphic: true
```
Comment (yakloinsteak) [inline app/models/login_activity.rb]: Will it complain when there's a failed login and no user? Might need `optional: true`

File: db/migrate/20190911124324_create_login_activities.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class CreateLoginActivities < ActiveRecord::Migration
+  def change
+    create_table :login_activities do |t|
+      t.string :scope
+      t.string :strategy
+      t.string :identity
```
Comment (yakloinsteak) [inline db/migrate/20190911124324_create_login_activities.rb]: I'd put `null: false` wherever I could, but I don't know if that breaks things.

File: Gemfile.lock [hunk 3]
```diff
@@ -74,6 +74,11 @@ GEM
     attr_encrypted (3.1.0)
       encryptor (~> 3.0.0)
     attribute_normalizer (1.2.0)
+    authtrail (0.1.3)
```
Comment (yakloinsteak) [inline Gemfile.lock]: the gem recommends [rack-attack](https://github.com/kickstarter/rack-attack) but that might be out of scope and definitely not required.

Comment (eanders) [reply Gemfile.lock]: Yeah, I noticed that.  Also of note, that would be good to dig into: https://ankane.org/hardening-devise and https://github.com/ankane/secure_rails

File: app/models/login_activity.rb (hunk 1 above)
Comment (eanders) [reply app/models/login_activity.rb]: Thanks, I'll add it.  We're still on Rails 4, but 5 is coming.

File: db/migrate/20190911124324_create_login_activities.rb (hunk 2 above)
Comment (eanders) [reply db/migrate/20190911124324_create_login_activities.rb]: Yeah, this is the default migration from the gem, so I'm going to leave it in this instance.

File: app/models/login_activity.rb (hunk 1 above)
Comment (eanders) [reply app/models/login_activity.rb]: Oddly, `optional: true` prevents the gem from working.  For now I'm leaving it as described in the docs. https://github.com/ankane/authtrail

**Pull request merged**